# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1290217458 18000
#      Fri Nov 19 20:44:18 2010 -0500
# Node ID 475e90eb4ff12fdd2001fbc9fcb073eede3710e5
# Parent  f83e17ec270dcd7adbf3896fd77b1a7a7b92098f
mex::free: don't warn about skipping memory not allocated by mx{M,C,Re}alloc

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-11-19  John W. Eaton  <jwe@octave.org>
+
+	* mex.cc (mex::free): Don't warn about skipping memory not
+	allocated by mxMalloc, mxCalloc, or mxRealloc.
+
 2010-11-18  Daisuke TAKAGO  <takago@neptune.kanazawa-it.ac.jp>
 
 	* gl-render.cc (opengl_renderer::draw_line):
 	Layer markers in 2D plots.
 
 2010-11-18  John W. Eaton  <jwe@octave.org>
 
 	Bug # 31689.
diff --git a/src/mex.cc b/src/mex.cc
--- a/src/mex.cc
+++ b/src/mex.cc
@@ -2211,18 +2211,20 @@ public:
             xfree (ptr);
           }
         else
           {
             p = foreign_memlist.find (ptr);
 
             if (p != foreign_memlist.end ())
               foreign_memlist.erase (p);
+#ifdef DEBUG
             else
               warning ("mxFree: skipping memory not allocated by mxMalloc, mxCalloc, or mxRealloc");
+#endif
           }
       }
   }
 
   // Mark a pointer to be freed on exit.
   void mark (void *ptr)
   {
 #ifdef DEBUG
